MixItUp.Desktop/Services/WindowsDeveloperAPIService.cs

```diff
@@ -71,300 +71,321 @@ public UserDeveloperAPIModel(UserDataViewModel userData)
-        public const string PutHttpMethod = "PUT";
-        public const string PatchHttpMethod = "PATCH";
+        public void Configuration(IAppBuilder appBuilder)
+        {
+            HttpConfiguration config = new HttpConfiguration();
+
+            config.Formatters.Clear();
+            config.Formatters.Add(new JsonMediaTypeFormatter());
+
+            config.MapHttpAttributeRoutes();
 
-        public const string DeveloperAPIHttpListenerServerAddress = "http://localhost:8911/api/";
+            appBuilder.UseWebApi(config);
+        }
+    }
+
+    public class WindowsDeveloperAPIService : IDeveloperAPIService
+    {
+        private IDisposable webApp;
+        public const string DeveloperAPIHttpListenerServerAddress = "http://localhost:8911/";
+
+        public bool Start()
+        {
+            // Ensure it is cleaned up first
+            End();
 
-        public WindowsDeveloperAPIService() : base(WindowsDeveloperAPIService.DeveloperAPIHttpListenerServerAddress) { }
+            this.webApp = WebApp.Start<WindowsDeveloperAPIServiceStartup>(DeveloperAPIHttpListenerServerAddress);
+            return true;
+        }
 
-        protected override async Task ProcessConnection(HttpListenerContext listenerContext)
+        public void End()
         {
-            if (!string.IsNullOrEmpty(listenerContext.Request.RawUrl))
+            if (this.webApp != null)
             {
-                string url = listenerContext.Request.RawUrl.ToLower();
-                string httpMethod = listenerContext.Request.HttpMethod;
-                if (url.StartsWith("/api"))
-                {
-                    List<string> urlSegments = new List<string>(url.Split(new string[] { "/" }, StringSplitOptions.RemoveEmptyEntries));
-                    urlSegments.RemoveAt(0);
-                    if (urlSegments.Count() == 0)
-                    {
-                        await this.CloseConnection(listenerContext, HttpStatusCode.OK, "Welcome to the Mix It Up Developer API! More detailed documentation about this service, please visit https://github.com/SaviorXTanren/mixer-mixitup/wiki/Developer-API");
-                        return;
-                    }
-                    else
-                    {
-                        string data = await this.GetRequestData(listenerContext);
-                        await this.ProcessDeveloperAPIRequest(listenerContext, httpMethod, urlSegments, data);
-                        return;
-                    }
-                }
+                this.webApp.Dispose();
+                this.webApp = null;
             }
-            await this.CloseConnection(listenerContext, HttpStatusCode.BadRequest, "This is not a valid API");
         }
+    }
 
-        private async Task ProcessDeveloperAPIRequest(HttpListenerContext listenerContext, string httpMethod, List<string> urlSegments, string data)
+    [RoutePrefix("api/mixer/users")]
+    public class MixerController : ApiController
```

Comment:
Wouldn't this be the MixerUsersController?

Reply:
Updated.